server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -613,7 +623,8 @@ private ClusterState applyCreateIndexRequestWithV2Template(
         final CreateIndexClusterStateUpdateRequest request,
         final boolean silent,
         final String templateName,
-        final BiConsumer<Metadata.Builder, IndexMetadata> metadataTransformer
+        final BiConsumer<Metadata.Builder, IndexMetadata> metadataTransformer,
+        boolean isSystemIndex
```

Comment:
Rather than passing this boolean around, I think we can cheat and update CreateIndexClusterStateUpdateRequest's settings to set a value for INDEX_REPLICATION_TYPE_SETTING earlier, only if the cluster setting is set to segment.

from normalizeRequestSetting method...
```java
     private void normalizeRequestSetting(CreateIndexClusterStateUpdateRequest createIndexClusterStateRequest) {
        Settings.Builder updatedSettingsBuilder = Settings.builder();
        final Settings.Builder builder = updatedSettingsBuilder.put(createIndexClusterStateRequest.settings())
            .normalizePrefix(IndexMetadata.INDEX_SETTING_PREFIX);
        if(CLUSTER_REPLICATION_TYPE_SETTING.get(this.settings).equals(ReplicationType.SEGMENT)
            && validateDotIndex(createIndexClusterStateRequest.index(), false)) {
            builder.put(INDEX_REPLICATION_TYPE_SETTING.getKey(), ReplicationType.DOCUMENT);
        }
        final Settings build = builder.build();
        indexScopedSettings.validate(build, true);
        createIndexClusterStateRequest.settings(build);
    }
```

wdyt?

Reply:
Sure, I have updated this in latest PR